theano/scan_module/scan_op.py

```diff
@@ -674,8 +674,8 @@ def make_thunk(self, node, storage_map, compute_map, no_recycling):
                   self.n_mit_sot +
                   self.n_sit_sot +
                   self.n_nit_sot)
-        wrapped_inputs = [Param(x, borrow=True) for x in self.inputs]
-        wrapped_outputs = [Out(x, borrow=False) for x in
+        wrapped_inputs = [Param(x, borrow=False) for x in self.inputs]
```

Comment:
To ensure there is not aliasing between the inputs and the outputs of the inner function. There are probably be cases where such aliasing would be without consequence but detecting these cases could be tricky. Worse case scenario when using this precaution is that some outputs of the inner function, that would otherwise be aliased, will have to be produced via DeepCopy but since we avoid the copy outside of the inner function, that doesn't really cost us anything.